File: 프로그래머스/1/12926. 시저 암호/시저 암호.py
def solution(s, n):
    answer = []
    
    for c in s:
        if c == ' ':
            answer.append(' ')
            continue
        
        offset = ord('a') if c.islower() else ord('A')
        position = (ord(c) - offset + n) % 26
        answer.append(chr(position + offset))
                
    return ''.join(answer)

File: 프로그래머스/1/12926. 시저 암호/시저 암호.java
class Solution {
    private char push(char c, int n) {
        if (!Character.isAlphabetic(c)) {
            return c;
        }
        
        int offset = Character.isLowerCase(c) ? 'a' : 'A';
        int position = (c - offset + n) % 26;
        
        return (char) (offset + position);
    }
    
    public String solution(String s, int n) {
        StringBuilder sb = new StringBuilder();
        
        for (char c : s.toCharArray()) {
            sb.append(push(c, n));
        }
        
        return sb.toString();
    }
}